app/core/store/modules/prepaids.js

```diff
@@ -269,66 +270,74 @@ export default {
+      const students = members
+        .map(data => data instanceof User ? data : new User(data))
+        .filter(u => u.isEnrolled() && u.prepaidType() === 'course')
 
       const existsLicenseToRevoke = students.length > 0
       if (!existsLicenseToRevoke) {
         noty({ text: `No licenses applied to selected student(s).` })
         return
       }
 
-      let confirmed = false
-      await new Promise((resolve) => noty({
-        text: `Revoking a license will make it available to apply to other students. Students will no longer be able to access paid content, but their progress will be saved. Please confirm you'd like to proceed.`,
-        buttons: [
-          {
-            addClass: 'btn btn-primary',
-            text: 'Ok',
-            onClick: function ($noty) {
-              confirmed = true
-              $noty.close()
-              resolve()
-            }
-          },
-          {
-            addClass: 'btn btn-danger',
-            text: 'Cancel',
-            onClick: function ($noty) {
-              $noty.close()
-              resolve()
+      if (!confirmed) {
+        await new Promise((resolve) => noty({
+          text: `Revoking a license will make it available to apply to other students. Students will no longer be able to access paid content, but their progress will be saved. Please confirm you'd like to proceed.`,
+          buttons: [
+            {
+              addClass: 'btn btn-primary',
+              text: 'Ok',
+              onClick: function ($noty) {
+                confirmed = true
+                $noty.close()
+                resolve()
+              }
+            },
+            {
+              addClass: 'btn btn-danger',
+              text: 'Cancel',
+              onClick: function ($noty) {
+                $noty.close()
+                resolve()
+              }
             }
-          }
-        ]
-      }))
+          ]
+        }))
 
-      if (!confirmed) {
-        return
+        if (!confirmed) {
+          return
+        }
       }
 
-      const promises = []
       for (const student of students) {
         const courseProducts = student.activeProducts('course')
-        courseProducts.map(product => {
-          let prepaid = new Prepaid({
+        await Bluebird.map(courseProducts, async product => {
+          const prepaid = new Prepaid({
             _id: product.prepaid,
             type: 'course'
           })
-          promises.push(new Promise((resolve, reject) =>
+          await new Promise((resolve, reject) =>
             prepaid.revoke(student, {
               success: resolve,
-              error: () => {
-                console.error(`Didn't revoke this license`)
-                resolve()
-              },
+              error: reject,
               data: { sharedClassroomId }
             })
-          ))
+          )
+
+          if (updateUserProducts) {
+            student.set(
```

Comment:
does this actually update backend or just client?

Reply:
not sure.. I've just moved the code from here: https://github.com/codecombat/codecombat/pull/6742/files#diff-9b2315837ffc0dd7f252fc763a82c2fad33c10ccf9ac802a06524bc27cff2492L830-L833

Reply:
Can u check revoke's backend code? I'm assuming in revoke API call backend already does it and that's why we set it in client here. Setting just in front-end doesn't make sense to me if change hasn't already been made it backend

Reply:
Haha ok. Though, I would prefer not to add a lot of new styles of code for similar things. Also, bluebird package hasn't been updated for 3 years so preferably better not to use it much!!